spec/03-types.md

```diff
@@ -102,16 +103,46 @@ forms.
 ### Singleton Types
 
 ```ebnf
-SimpleType  ::=  Path ‘.’ type
+SimpleType  ::=  Path ‘.’ ‘type’
 ```
 
-A _singleton type_ is of the form $p.$`type`, where $p$ is a
-path pointing to a value expected to [conform](06-expressions.html#expression-typing)
-to `scala.AnyRef`. The type denotes the set of values
-consisting of `null` and the value denoted by $p$.
+A _singleton type_ is of the form $p.$`type`. Where $p$ is a path pointing to a
+value which [conforms](06-expressions.html#expression-typing) to
+`scala.AnyRef`, the type denotes the set of values consisting of `null` and the
+value denoted by $p$ (i.e., the value $v$ for which `v eq p`). Where the path
+does not conform to `scala.AnyRef` the type denotes the set consisting of only
+the value denoted by $p$.
 
-A _stable type_ is either a singleton type or a type which is
-declared to be a subtype of trait `scala.Singleton`.
+<!-- a pattern match/type test against a singleton type `p.type` desugars to `_ eq p` -->
+
+### Literal Types
+
+```ebnf
+SimpleType  ::=  Literal
+```
+
+A literal type `lit` is a special kind of singleton type which denotes the
+single literal value `lit`.  Thus, the type ascription `1: 1` gives the most
+precise type to the literal value `1`:  the literal type `1`.
+
+At run time, an expression `e` is considered to have literal type `lit` if `e ==
+lit`.  Concretely, the result of `e.isInstanceOf[lit]` and `e match { case _ :
+lit => }` is determined by evaluating `e == lit`.
+
+Literal types are available for all types for which there is dedicated syntax
+except `Unit`. This includes the numeric types (other than `Byte` and `Short`
+which don't currently have syntax), `Boolean`, `Char`, `String` and `Symbol`.
+
+<!-- TODO: use eq when we lift it up to Any -->
```

Comment:
I'll tweak the wording.

Reply:
Looking at this now ... those TODOs are very stale and I think can be deleted. The one you're flagging up here relates to the specification being updated to use `eq` rather than `==` in the definition of `isInstanceOf` for literal types _when_ `eq` is defined for values of type `Any`. I don't think this is on the table at the moment. The following comment about `Byte` relates to a hypothetical future SIP which adds `Byte` and `Short` literals ... this is mentioned in SIP 23 in the section "Follow on work from this SIP" and I don't think needs to be mentioned as a comment in the spec source. The final comment relates to inferred constant types and that's covered in the local type inference section of the spec.